File: aes-128-cbc.js
var crypto = require('crypto'),
      algorithm = 'aes-128-cbc',
      password = 'example no key death.';

function encrypt(text){
  var cipher = crypto.createCipher(algorithm,password)
  var crypted = cipher.update(text,'utf8','hex')
  crypted += cipher.final('hex');
  return crypted;
}

function decrypt(text){
  var decipher = crypto.createDecipher(algorithm,password)
  var dec = decipher.update(text,'hex','utf8')
  dec += decipher.final('utf8');
  return dec;
}

var encrypted = encrypt("example no plain text death.")
console.log(encrypted);
console.log(decrypt(encrypted));

File: aes-128-cbc.py
import os
import hashlib

from Crypto.Cipher import AES

def padding(data):
  pad = AES.block_size - (len(data) % AES.block_size) + len(data)
  out = data.ljust(pad, ' ')
  return out

def encrypt(key, iv, plain_text):
  data = padding(plain_text)
  encryptor = AES.new(key, AES.MODE_CBC, iv)
  encrypted = encryptor.encrypt(data)
  return encrypted

def decrypt(key, iv, encrypted):
  decryptor = AES.new(key, AES.MODE_CBC, iv)
  decrypted = decryptor.decrypt(encrypted)
  return decrypted

def main():
  key = hashlib.sha256('example no key death.').digest()
  iv = os.urandom(AES.block_size)

  plain_text = 'example no plain text death.'
  encrypted = encrypt(key, iv, plain_text)
  decrypted = decrypt(key, iv, encrypted)

  print 'encrypt: %s' % (encrypted.encode('hex'))
  print 'decrypt: %s' % (decrypted)

if __name__ == "__main__":
  main()
